### ISSUE TITLE
Create a nonprivileged user

### ISSUE THREAD

**Issue by tomassedovic** [imain/ocp-doit#3]
Every command including the OpenShift deployment is run as the OpenStack admin user / project. This is neither necessary, nor desirable (the installer should work for regular cloud users and therefore we should test it as such.

The script below creates a normal user with one extra privilege (we do have to add the `swiftoperator` role to be able to create containers).

It expects two `clouds.yaml` entries: `standalone-admin` (the admin user, currently called `standalone` in ocp-doit) and `standalone` -- the new user we're going to be creating.

This creates the necessary networks, security group rules and so on to be able to run openshift-installer. I have tested it and it all seems to be working.

I suggest we change ocp-doit to always do this and run the installer as a regular user, but that would require changes to clouds.yaml and a bunch of the scripts so I'm opening this issue first. Maybe we just want to ship this script instead and not disrupt the existing ones.

```bash
#!/bin/bash

set -eux
set -o pipefail

# Create & configure the user, project and quota
export OS_CLOUD=standalone-admin

openstack project create openshift
openstack user create --password 'password' openshift
openstack role add --user openshift --project openshift _member_
openstack role add --user openshift --project openshift swiftoperator
# NOTE(shadower): minimum secgroup: 6, secgroup-rules: 60
openstack quota set --secgroups 100 --secgroup-rules 1000 openshift

# Set the user's key, router, subnet, tempurl access, etc.
export OS_CLOUD=standalone

openstack keypair create --public-key ~/.ssh/id_rsa.pub default
openstack network create --internal private
openstack subnet create private --subnet-range 192.168.24.0/24 --network private
openstack router create internets
openstack router set internets --external-gateway public
openstack router add subnet internets private
openstack object store account set --property Temp-URL-Key=superkey

openstack security group rule create default --protocol tcp --dst-port 22:22 --remote-ip 0.0.0.0/0
openstack security group rule create --protocol icmp default
openstack security group rule create --protocol udp --dst-port 53:53 default
```

**Comment by flaper87** [comment]
I like the idea of developing as you'd expect ppl to use the project. I'm all for switching ocp-doit to always create a non-privileged user and run commands as that user. Standalone deployment does require sudo, tho but it's the exception rather than the norm.

**Comment by tomassedovic** [comment]
Yeah. This is mostly about the OpenStack user itself. And personally, I deploy openshift from my laptop, using standalone as an external cloud (so no sudo required there).

I'll create a PR.

**Comment by trown** [comment]
+1 seems like a good improvement. Thanks Tomas.
